sky/serve/autoscalers.py

```diff
@@ -318,12 +320,51 @@ def get_latest_version_with_min_replicas(
     def select_outdated_replicas_to_scale_down(
             self, replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
+        """Select outdated replicas to scale down."""
+
+        if self.update_mode == serve_utils.UpdateMode.ROLLING:
+            latest_ready_replicas = []
+            old_nonterminal_replicas = []
+            for info in replica_infos:
+                if info.version == self.latest_version:
+                    if info.is_ready:
+                        latest_ready_replicas.append(info)
+                elif not info.is_terminal:
+                    old_nonterminal_replicas.append(info)
+
+            num_latest_ready_replicas = len(latest_ready_replicas)
+
+            # We compare to target_num_replicas instead of min_replicas, to
+            # guarantee better service quality. Since mixing traffic across
+            # old and latest versions are allowed in rolling update, this will
+            # not affect the time it takes for the service to updated to the
+            # latest version.
+            if num_latest_ready_replicas >= self.target_num_replicas:
+                # Once the number of ready new replicas is greater than or equal
+                # to the target, we can scale down all old replicas.
+                return [info.replica_id for info in old_nonterminal_replicas]
```

Comment:
Do we need to only scale down old non-terminal replicas?
Say on version 1 I saw a failed replica, I fix something and update to a second version, I might not want to see the failed replica (version 1) in my status.